lnwallet/channel.go

```diff
@@ -6058,41 +6060,45 @@ func (lc *LightningChannel) validateFeeRate(feePerKw SatPerKWeight) error {
 	// Using the weight of the commitment transaction if we were to create
 	// a commitment now, we'll compute our remaining balance if we apply
 	// this new fee update.
-	newFee := lnwire.NewMSatFromSatoshis(
-		feePerKw.FeeForWeight(txWeight),
+	newFee := lnwire.NewMSatFromSatoshis(feePerKw.FeeForWeight(txWeight))
+	maxFee := lnwire.NewMSatFromSatoshis(
+		btcutil.Amount(float64(baseBalance) * maxFeeAllocation),
 	)
+	balanceAfterFee := baseBalance - newFee
+
+	switch {
+	// The new fee should not exceed our maximum fee allowed.
+	case newFee > maxFee:
+		return fmt.Errorf("cannot apply fee_update=%v sat/kw, new fee "+
```

Comment:
I think we may actually just need to cooperative close at this point, as otherwise the channel won't be able to proceed as is, since we don't (but have some cut out design space for) have the ability to mark rejected updates explicitly. If we return an error here, we won't apply the fee, and the channel will be force closed by the receiving party. 

What I had in mind with this PR was that we would _refrain_ from triggering a new fee update if thee sampled fee update was above some threshold. This results in no force closures as we simply regulate our behavior on the end of the initiator. Within the protocol we're missing the ability for the _responder_ to hav some say w.r.t what the final fee is, as they also want to ensure that they're able to close the channel in timely manner, and also this fee serves as the upper bound of what the fee can be for the cooperative closure transaction.

Reply:
> I think we may actually just need to cooperative close at this point, as otherwise the channel won't be able to proceed as is, since we don't (but have some cut out design space for) have the ability to mark rejected updates explicitly.

From a previous discussion offline, it seemed like we indeed want to do this but later down the road?

> What I had in mind with this PR was that we would refrain from triggering a new fee update if thee sampled fee update was above some threshold. This results in no force closures as we simply regulate our behavior on the end of the initiator.

Wasn't aware of the force close issue. The fee update is now not proposed if it exceeds the max.

Reply:
> From a previous discussion offline, it seemed like we indeed want to do this but later down the road?

Yep, the goal of this is just to ensure we don't bleed away our entire channel. We can possibly add a flag in this PR to _optionally_ close them out in order to avoid some fee stampede that results in a large number of channels being closed.